packages/core/core/prompt/attachments-to-parts.ts

```diff
@@ -0,0 +1,74 @@
+import { Attachment } from '@ai-sdk/ui-utils';
+import { ImagePart, TextPart } from './content-part';
+import { convertDataContentToUint8Array } from './data-content';
+import { getErrorMessage } from '@ai-sdk/provider-utils';
+
+type ContentPart = TextPart | ImagePart;
+
+/**
+ * Converts a data URL of type text/* to a text string.
+ */
+export function dataUrlToText(dataUrl: string): string {
+  return Buffer.from(dataUrl, 'base64').toString('utf-8');
```

Comment:
does this work in a browser env?

Reply:
https://github.com/vercel/ai/pull/2226#discussion_r1678314642